src/wp-admin/options-general.php

```diff
@@ -97,7 +97,76 @@
 <p class="description" id="tagline-description"><?php echo $tagline_description; ?></p></td>
 </tr>
 
+<tr class="hide-if-no-js">
+<th scope="row"><?php _e( 'Site Icon' ); ?></th>
+<td>
+	<?php
+
+	wp_enqueue_media();
+	wp_enqueue_script( 'site-icon' );
+
+	$classes_for_upload_button = 'upload-button button-add-media button-add-site-icon';
+	$classes_for_update_button = 'button';
+
+	$classes_for_avatar = 'avatar avatar-150';
+	if ( has_site_icon() ) {
+		$classes_for_avatar          .= ' has-site-icon';
+		$classes_for_button           = $classes_for_update_button;
+		$classes_for_button_on_change = $classes_for_upload_button;
+	} else {
+		$classes_for_avatar          .= ' hidden';
+		$classes_for_button           = $classes_for_upload_button;
+		$classes_for_button_on_change = $classes_for_update_button;
+	}
+
+
+	?>
+
+	<img id="site-icon-img" class="<?php echo esc_attr( $classes_for_avatar ); ?>" src="<?php site_icon_url( null, 150 ); ?>" height="150" width="150" alt="<?php esc_attr_e( 'Preview as a browser icon' ); ?>" />
```

Comment:
As far as I see the image preview is just the image. There's no visual representation of a browser tab like it used to be in the Customizer. That's totally fine but I'm wondeering whether the alt text `Preview as a browser icon` is appropriate then, as there's no 'browser' thing in the preview.

More importantly, it would be great to provide screen reader users with some information about _what_ the actual image is. Since this is an authoring context, the image `alt` text wouldn't be entirely appropriate. In a few places, for example the media widgets or the Gutenberg featured image UI, there's a pattern in use that either uses a sentence with the alt text in it or, if the alt is empty, another sentence with the image file name. See an example here:

https://github.com/WordPress/gutenberg/blob/45a49c9624e551b28592e4d7baed75bdf79fed6e/packages/editor/src/components/post-featured-image/index.js#L137-L151

```
__( 'Current image: %s' )
or
__( 'The current image has no alternative text. The file name is: %s' )
```

Reply:
I updated this to use the same UI as the cusomizer for previewing, so `Preview as a browser icon` is accurate now.

Reply:
The customizer isn't the best example of accessibility. I'm not sure this addresses the concern I reported. As of now, there's no textual information about the currently selected image. As such, screen reader users won't have a clue what the site icon image is, thus the whole feature is barely usable for them.